src/otx/algo/detection/huggingface_model.py

```diff
@@ -0,0 +1,195 @@
+            pretrained_model_name_or_path=self.model_name,
+            num_labels=num_classes,
+            ignore_mismatched_sizes=True,
+        )
+
+    def _customize_inputs(
+        self,
+        entity: DetBatchDataEntity,
+        pad_size_divisor: int = 32,
+        pad_value: int = 0,
+    ) -> dict[str, Any]:
+        labels = [{"class_labels": entity.labels[i], "boxes": entity.bboxes[i]} for i in range(entity.batch_size)]
+
+        if isinstance(entity.images, list):
+            entity.images, entity.imgs_info = stack_batch(
+                entity.images,
+                entity.imgs_info,
+                pad_size_divisor=pad_size_divisor,
+                pad_value=pad_value,
+            )
+
+        return {
+            "pixel_values": entity.images,
+            "labels": labels,
+        }
+
+    def _customize_outputs(
+        self,
+        outputs: ModelOutput,
+        inputs: DetBatchDataEntity,
+    ) -> DetBatchPredEntity | OTXBatchLossEntity:
+        if self.training:
+            return outputs.loss_dict
+
+        target_sizes = torch.tensor([box.canvas_size for box in inputs.bboxes])
+        results = self.image_processor.post_process_object_detection(
+            outputs,
+            0.0,
+            target_sizes=target_sizes,
+        )
+
+        scores, labels, bboxes = [], [], []
+        for i, result in enumerate(results):
+            scores.append(result["scores"])
+            labels.append(result["labels"])
+            bboxes.append(
+                tv_tensors.BoundingBoxes(
+                    result["boxes"],
+                    format="XYXY",
+                    canvas_size=inputs.bboxes[i].canvas_size,
+                ),
+            )
+
+        if self.explain_mode:
+            msg = "Explain mode is not supported yet."
+            raise NotImplementedError(msg)
+
+        return DetBatchPredEntity(
+            batch_size=inputs.batch_size,
+            images=inputs.images,
+            imgs_info=inputs.imgs_info,
+            scores=scores,
+            labels=labels,
+            bboxes=bboxes,
+        )
+
+    @property
+    def _exporter(self) -> OTXModelExporter:
+        """Creates OTXModelExporter object that can export the model."""
+        image_size = (1, 3, *self.image_processor.size.values())
+        image_mean = (0.0, 0.0, 0.0)
+        image_std = (255.0, 255.0, 255.0)
+
+        return OTXNativeModelExporter(
+            task_level_export_parameters=self._export_parameters,
+            input_size=image_size,
+            mean=image_mean,  # type: ignore[arg-type]
+            std=image_std,  # type: ignore[arg-type]
+            resize_mode="standard",
+            swap_rgb=False,
+            via_onnx=False,
+            onnx_export_configuration={
+                "input_names": ["images"],
+                "output_names": ["bboxes", "labels", "scores"],
+                "dynamic_axes": {
+                    "images": {0: "batch", 2: "height", 3: "width"},
```

Comment:
Is that possible to avoid dynamic height and width?

Reply:
i removed that. thanks.https://github.com/openvinotoolkit/training_extensions/pull/3747/commits/3b01466ec685a147eefefc016e4a882fb2edd906